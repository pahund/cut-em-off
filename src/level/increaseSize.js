const connections = {
    0: [0, 0],
    1: [11, 3],
    2: [0, 3],
    3: [0, 3],
    4: [11, 0],
    5: [11, 3],
    6: [11, 3],
    7: [0, 3],
    8: [0, 0],
    9: [11, 0],
    10: [0, 0],
    11: [11, 0],
    12: [11, 3],
    13: [0, 3],
    14: [0, 0],
    15: [0, 0],
    16: [11, 0]
};

export default (width, mapData) =>
    mapData
        .reduce((acc, tile) => [...acc, tile, connections[tile][0]], [])
        .reduce(
            (acc, tile, index, arr) =>
                index % 2 > 0
                    ? acc
                    : acc === null
                        ? [...arr.slice(0, width * 2), connections[tile][1], 0, ...arr.slice(width * 2)]
                        : [
                              ...acc.slice(0, width * 2 + index),
                              connections[tile][1],
                              0,
                              ...acc.slice(width * 2 + index)
                          ],
            null
        );
