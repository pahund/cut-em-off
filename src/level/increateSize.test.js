import increaseSize from './increaseSize.js';

test.each([
    // [4, [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16]],
    [2, [1, 2, 9, 10]]
])('When I call “increaseSize” with width %j and values %j, the result is correct', (width, input) =>
    expect(increaseSize(width, input)).toMatchSnapshot()
);
