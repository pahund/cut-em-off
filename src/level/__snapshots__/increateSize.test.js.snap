// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When I call “increaseSize” with width 2 and values [1,2,9,10], the result is correct 1`] = `
Array [
  0,
  0,
  0,
  0,
  0,
  3,
  11,
  0,
  1,
  11,
  3,
  2,
  0,
  9,
  11,
  10,
  0,
]
`;
